### ISSUE TITLE
Infinity loop in those bus routes whats contain a bus stops at twice

### ISSUE THREAD

**Issue by summoner001** [osmandapp/OsmAnd#15449]
# 🐞 routing report

### Routing engine

- [ x] OsmAnd's in-app offline routing
- [ ] Any online routing provider (YOURS, openrouteservice, OSRM, etc.)

### Routing Profile

 Browsing

### Start and end points

The bus route: https://www.openstreetmap.org/relation/14616287#map=14/47.2179/21.5378

If I open the bus route in OsmAND and stepping the stations with the next button, OsmAND get into an infinite loop if a bus route includes a bus stop at twice, even though the timetable actually says it stops at that bus stop at twice in a single outward journey.

1st bus stop => 2nd bus stop => 3rd bus stop => ... => 9th bus stop (Berettyóújfalu, bus station) (first) => 10th bus stop (hospital) => 11th bus stop (Berettyóújfalu, bus station) (second, again) and at this point osmand don't jump to 12th bus stop when i click the next button. Osmand get into trouble and infinite loop 10th bus stop <=> 11th bus stop he never jump to 12th bus stop.

The bus stops are not incorrectly included twice in the route. This is the correct timetable.

https://user-images.githubusercontent.com/11212632/194753991-69709648-b2da-441c-82aa-93f7724c67ec.mp4

### Is this a regression?

 No

## 🌍 Your Environment

**OsmAnd Version: v4.2.7 (F-Droid)**

<pre><code>

</code></pre>

**Device and Android/iOS version:**
Android 9
**Maps used (online or offline):**

- [ x] Offline maps offered within the OsmAnd app for download.

- [ ] Online (tile / raster) maps 

**Anything else relevant?**

**Comment by Max1234-Ita** [comment]
Hi, could it be similar to this?
https://github.com/osmandapp/OsmAnd/issues/15013

I just stumbled on it while searching for a different issue and I thought these can be related, feel free to correct me, if I'm wrong.

**Comment by summoner001** [comment]
I think the two problems are not the same. 
#15013 Navigation has a problem with route search. Or it can be caused by a wrong mandatory direction of travel at an intersection. I have encountered such an error and corrected it in osm.

But in this issue, however, the bus route viewer cannot identify the bus stop if it appears twice in the route. The viewer thinks it is there for the first time. But it is the second time.

**Comment by ivanPyrohivskyi** [comment]
[Stop Berettyóújfalu, autóbusz-állomás](https://www.openstreetmap.org/node/10015293348) is present in [Berettyószentmárton => Berettyóújfalu TESCO](https://www.openstreetmap.org/relation/14616287) twice.

**Comment by summoner001** [comment]
> [Stop Berettyóújfalu, autóbusz-állomás](https://www.openstreetmap.org/node/10015293348) is present in [Berettyószentmárton => Berettyóújfalu TESCO](https://www.openstreetmap.org/relation/14616287) twice.

Yes the bus stop present in the route at twice. But this is the bus's timetable. The bus stopping at twice in the route. This is not problem!

**Comment by ivanPyrohivskyi** [comment]
Need to fix in iOS also

**Comment by alex-osm** [comment]
https://github.com/osmandapp/OsmAnd-core/commit/5e1a1aa6cb385f3881ede6e56069dd25fe807201
https://github.com/osmandapp/OsmAnd-iOS/commit/084801fd5e0563c0fa1dc220b5b88ddfd7b571cf

**Comment by DmitryAlexei** [comment]
OsmAnd~ 4.6.0#489m, released: 2023-07-26

https://github.com/osmandapp/OsmAnd/assets/116079541/f69114ab-8fd0-462b-899d-49fea11b13d9
